Contribute/how-to-write-use-markdown.md

```diff
@@ -383,10 +383,329 @@ You can see an example of selectors in action at the [Azure docs](https://docs.m
+[aspnetcore/data/ef-mvc/intro/samples/cu/Controllers/StudentsController.cs](https://github.com/aspnet/Docs/blob/master/aspnetcore/data/ef-mvc/intro/samples/cu/Controllers/StudentsController.cs)
+in the same repository.
+
+#### Selected line numbers
+
+Example:
+
+```markdown
+:::code language="csharp" source="intro/samples/cu/Controllers/StudentsController.cs" range="2-24,26":::
+```
+
+This example displays only lines 2-24 and 26 of the *StudentController.cs* code file.
+
+Prefer snippets over hard-coded line numbers, as explained in the next section.
+
+#### Named snippet
+
+Example:
+
+```markdown
+:::code language="csharp" source="intro/samples/cu/Controllers/StudentsController.cs" id="snippet_Create":::
+```
+
+Use only letters and underscores for the name.
+
+The example displays the `snippet_Create` section of the code file. The code file for this example
+has a C# region named `snippet_Create`:
+
+```cs
+// code excluded from the snippet
+// <snippet_Create>
+// code included in the snippet
+// </snippet_Create>
+// code excluded from the snippet
+```
+
+Whenever you can, refer to a named section rather than specifying line numbers. Line number
+references are brittle because code files inevitably change in ways that make line numbers change.
+You don't necessarily get notified of such changes. Your article eventually starts showing the wrong
+lines and you have no clue anything has changed.
+
+#### Highlighting selected lines
+
+Example:
+
+```markdown
+:::code language="csharp" source="intro/samples/cu/Controllers/StudentsController.cs" range="2-24,26" highlight="2,5":::
+```
+
+The example highlights lines 2 and 5, counting from the start of the displayed snippet. (Line
+numbers to highlight don't count from the start of the code file.) In other words, lines 3 and 6
+of the code file are highlighted.
+
+#### Interactive code snippets
+
+You can enable interactive mode for code snippets included by reference. Here are examples:
+
+```markdown
+:::code source="PowerShell.ps1" interactive="cloudshell-powershell":::
+```
+
+```markdown
+:::code source="Bash.sh" interactive="cloudshell-bash":::
+```
+
+To turn on this feature for a particular code block, use the `interactive` attribute. The
+available attribute values are:
+
+- `cloudshell-powershell` - Enables the Azure PowerShell Cloud Shell, as in the preceding
+  example
+- `cloudshell-bash` - Enables the Azure Cloud Shell
+- `try-dotnet` - Enables Try .NET
+- `try-dotnet-class` - Enables Try .NET with class scaffolding
+- `try-dotnet-method` - Enables Try .NET with method scaffolding
+
+For the Azure Cloud Shell and PowerShell Cloud Shell, users can run commands against only their own
+Azure account.
+
+For Try .NET, there are three options for interactivity: `try-dotnet`, `try-dotnet-class`, and `try-dotnet-method`. Use of these options require no additional configuration within the code snippet. The default usings that are being used are:
```

Comment:
Consider pointing out right up front that Try .NET is for C#.  Folks will catch on quicker it Try.NET applies to their situation.  Also, it would be good to know what it is best used for and app examples where you might not.  Maybe link out to those types of details elsewhere?  I love that Try .NET info is getting integrated in here.  Thanks!!!

Reply:
Is "usings" here an actual term used for Try.NET?   Would it be more accurate to change this: "The default usings that are being used are:"  to "The namespaces currently supported and available by default are:" (Assuming here that the currently supported namespaces are exactly the same list available by default.)  Or perhaps change to:  "The namespaces currently available by default are:"  (Assuming in this case that more namespaces are supported than is currently available by default.)

Reply:
@wadepickett just updated for ya!